### PR TITLE
Added compatability with PHPUnit8

### PR DESCRIPTION
**Pull request by AndreyChorniy**
### Description (*)
Added compatability with PHPUnit8 for module Magento_Authorization

### Fixed Issues (if relevant)

1. magento/magento2#27500

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Authorization/Test/Unit/Model/Acl/AclRetrieverTest.php [hunk 1]
```diff
@@ -7,26 +7,36 @@
...
+    private $aclRetriever;
 
-    /** @var \PHPUnit_Framework_MockObject_MockObject|Role $roleMock */
-    protected $roleMock;
+    /**
+     * @var Role|MockObject
+     */
+    private $roleMock;
 
     protected function setup()
```
Comment (dmytro-ch) [inline app/code/Magento/Authorization/Test/Unit/Model/Acl/AclRetrieverTest.php]:
```suggestion
    protected function setUp(): void
```

File: app/code/Magento/Authorization/Test/Unit/Model/CompositeUserContextTest.php [hunk 2]
```diff
@@ -6,32 +6,36 @@
...
-    protected $compositeHelperMock;
+    private $compositeHelperMock;
 
     /**
      * @var ObjectManager
      */
-    protected $objectManager;
+    private $objectManager;
 
     protected function setUp()
```
Comment (dmytro-ch) [inline app/code/Magento/Authorization/Test/Unit/Model/CompositeUserContextTest.php]:
```suggestion
    protected function setUp(): setUp
```

File: app/code/Magento/Authorization/Test/Unit/Model/CompositeUserContextTest.php [hunk 3]
```diff
@@ -6,32 +6,36 @@
```
Comment (dmytro-ch) [inline app/code/Magento/Authorization/Test/Unit/Model/CompositeUserContextTest.php:7]:
```suggestion
declare(strict_types=1);

```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hello @AndreyChorniy, thank you for your contribution!
Unfortunately, the compatibility issues still present.
```
PHP Fatal error:  Declaration of Magento\Authorization\Test\Unit\Model\Acl\AclRetrieverTest::setup() must be compatible with PHPUnit\Framework\TestCase::setUp(): void in /home/dev/sites/bravo/app/code/Magento/Authorization/Test/Unit/Model/Acl/AclRetrieverTest.php on line 181
```
Please make sure you test your chanes with appropriate PHPUnit version.
Could you please also switch to the strict typing for all other existing methods within the test classes under refactoring?

Thank you!

File: app/code/Magento/Authorization/Test/Unit/Model/Acl/AclRetrieverTest.php (hunk 1 above)
Comment (AndreyChorniy) [reply app/code/Magento/Authorization/Test/Unit/Model/Acl/AclRetrieverTest.php]: Hi @dmytro-ch, 
Thank you for your review. 
I have fixed all issues. 
Could you please see my PR once again. 
Thank you.

File: app/code/Magento/Authorization/Test/Unit/Model/CompositeUserContextTest.php (hunk 2 above)
Comment (AndreyChorniy) [reply app/code/Magento/Authorization/Test/Unit/Model/CompositeUserContextTest.php]: I have fixed this.

File: app/code/Magento/Authorization/Test/Unit/Model/CompositeUserContextTest.php (hunk 3 above)
Comment (AndreyChorniy) [reply app/code/Magento/Authorization/Test/Unit/Model/CompositeUserContextTest.php:7]: Declaration has been added.

Comment (AndreyChorniy) [reply app/code/Magento/Authorization/Test/Unit/Model/CompositeUserContextTest.php:7]: Strict typing declaration has been added to each test script.

**Review by lbajsarowicz — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7268](https://jira.corp.magento.com/browse/ENGCOM-7268) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by dmytro-ch — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-7268](https://jira.corp.magento.com/browse/ENGCOM-7268) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
QA not applicable

**Comment by engcom-Echo** [comment]
@magento run Functional Tests B2B

**Review by slavvka — CHANGES_REQUESTED** [review]
Please resovle conflicts

**Comment by AndreyChorniy** [comment]
Conflicts have been resolved.

**Review by slavvka — APPROVED** [review]

**Pull request merged**